src/endpoint_resolver.rs

```diff
@@ -83,10 +82,8 @@ impl MomentoEndpointsResolver {
     }
 
     fn hosted_zone_endpoint(hosted_zone: Option<String>, prefix: &str) -> Option<MomentoEndpoint> {
-        if hosted_zone.is_none() {
-            return None;
-        }
-        let hostname = format!("{}{}", prefix, hosted_zone.clone().unwrap());
+        hosted_zone.as_ref()?;
+        let hostname = format!("{}{}", prefix, hosted_zone.unwrap());
         Some(MomentoEndpointsResolver::wrapped_endpoint(
             prefix, hostname, "",
         ))
```

Comment:
probably a more rusty way to option-coalesce would be something like 
```suggestion
fn hosted_zone_endpoint(hosted_zone: Option<String>, prefix: &str) -> Option<MomentoEndpoint> {
        hosted_zone.map(|hosted_zone| {
            let hostname = format!("{}{}", prefix, hosted_zone);
            MomentoEndpointsResolver::wrapped_endpoint(prefix, hostname, "")
        })
```

which avoids unwrap() and is shorter. The `.map` thing on Option is pretty nice https://doc.rust-lang.org/std/option/enum.Option.html#method.map